servlet/src/main/scala/org/http4s/servlet/ServletIo.scala

```diff
@@ -113,71 +111,69 @@ final case class NonBlockingServletIo[F[_]: Effect](chunkSize: Int) extends Serv
         // This effect sets the callback and waits for the first bytes to read
         val registerRead =
           // Shift execution to a different EC
-          Async.shift(Trampoline) *>
-            F.async[Option[Chunk[Byte]]] { cb =>
-              if (!state.compareAndSet(Init, Blocked(cb)))
-                cb(Left(bug("Shouldn't have gotten here: I should be the first to set a state")))
-              else
-                in.setReadListener(
-                  new ReadListener {
-                    override def onDataAvailable(): Unit =
-                      state.getAndSet(Ready) match {
-                        case Blocked(cb) => read(cb)
-                        case _ => ()
-                      }
+          F.async_[Option[Chunk[Byte]]] { cb =>
```

Comment:
Does this also take care of shifting?

Reply:
@rossabaker From what I understood by reading [this](https://github.com/typelevel/cats-effect/tree/series/3.x#ergonomics), shifting happens automatically. Would be great if you could confirm that?

Reply:
I'm trying to recall why that had to trampoline before.  I think `evalOn` would run the effect on a specified EC and then shift back.  This shifted once and never back, which I don't see a way to do without `ContextShift`.

I'm content to try without this and see if it works.